PEP8 linting by Prospector is not accurate

**Describe the bug**
The current prospector operation does not catch PEP8 errors in the code.

**To Reproduce**
Steps to reproduce the behavior:
```
(terndev) nisha@ubuntu:~/terndev/tern$ pycodestyle .
./setup.py:11:1: E302 expected 2 blank lines, found 1
./setup.py:15:1: E302 expected 2 blank lines, found 1
./setup.py:20:1: E305 expected 2 blank lines after class or function definition, found 1
./setup.py:26:73: W291 trailing whitespace
./setup.py:27:5: E128 continuation line under-indented for visual indent
./setup.py:31:14: E251 unexpected spaces around keyword / parameter equals
./setup.py:31:16: E251 unexpected spaces around keyword / parameter equals
./tern/report/errors.py:21:80: E501 line too long (81 > 79 characters)
./tern/report/report.py:178:80: E501 line too long (101 > 79 characters)
./tern/command_lib/command_lib.py:34:80: E501 line too long (80 > 79 characters)
./tern/utils/container.py:68:9: E128 continuation line under-indented for visual indent
./tern/utils/container.py:96:80: E501 line too long (113 > 79 characters)
./tern/utils/dockerfile.py:32:3: W605 invalid escape sequence '\$'
./tern/utils/dockerfile.py:32:5: W605 invalid escape sequence '\{'
./tern/utils/dockerfile.py:32:7: W605 invalid escape sequence '\}'
./tern/utils/rootfs.py:178:17: E128 continuation line under-indented for visual indent
./tern/classes/docker_image.py:166:80: E501 line too long (81 > 79 characters)
./tern/classes/package.py:151:80: E501 line too long (80 > 79 characters)
./tests/test_class_docker_image.py:28:80: E501 line too long (85 > 79 characters)
./tests/test_class_docker_image.py:29:20: E128 continuation line under-indented for visual indent
./ci/evaluate_docs.py:17:56: W291 trailing whitespace
./ci/evaluate_docs.py:28:8: E225 missing whitespace around operator
./ci/evaluate_docs.py:39:29: W291 trailing whitespace
./ci/test_files_touched.py:49:40: E231 missing whitespace after ':'
```
pycodestyle is a dependency for Prospector, however Prospector doesn't seem to be using it.

**Error in terminal**
See above

**Expected behavior**
Prospector should run `pycodestyle` and catch such PEP8 errors.

**Environment you are running Tern on**
Ubuntu 18.04 LTS VM

Prospector also doesn't accurately catch all the pep8 errors that pycodestyle catches
```
(terndev) nisha@ubuntu:~/terndev/tern$ prospector setup.py
Messages
========

setup.py
  Line: 20
    pep8: E305 / expected 2 blank lines after class or function definition, found 1 (col 1) <-- only line 20
```
vs.
```
(terndev) nisha@ubuntu:~/terndev/tern$ pycodestyle setup.py
setup.py:11:1: E302 expected 2 blank lines, found 1
setup.py:15:1: E302 expected 2 blank lines, found 1
setup.py:20:1: E305 expected 2 blank lines after class or function definition, found 1
setup.py:26:73: W291 trailing whitespace
setup.py:27:5: E128 continuation line under-indented for visual indent
setup.py:31:14: E251 unexpected spaces around keyword / parameter equals
setup.py:31:16: E251 unexpected spaces around keyword / parameter equals
```

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -25,11 +25,7 @@ jobs:
       - setup
       - run: pip install -r dev-requirements.txt
       - run: pip install .
-      - run: python ci/evaluate_docs.py
-      - run:
-         name: Prospector
-         when: on_success
-         command: prospector .
+      - run: python ci/evaluate_docs.py | xargs prospector
   # security linting using Bandit
   security:
     executor: ubuntu1604
@@ -37,7 +33,7 @@ jobs:
     steps:
       - setup
       - run: pip install -r dev-requirements.txt
-      - run: bandit -r .
+      - run: python ci/evaluate_docs.py | xargs bandit
   # linting for PR commit messages
   commit_check:
     executor: ubuntu1604

--- a/ci/evaluate_docs.py
+++ b/ci/evaluate_docs.py
@@ -5,18 +5,10 @@
 
 from git import Repo
 import os
-import sys
 
-
-# This script is designed to be run with circleci.
-# This script should pass if we need to run Bandit and Prospector,
-# which means that there are files other than *.md files being changed.
-# If only *.md files are changed, we don't need to run Bandit
-# or Prospector and the script will fail.
-
-# Assume that all files are *.md until proven otherwise
-docs_only = True
-
+# This is meant to run within circleci
+# Print out only .py files that have changed
+# Pipe to any linting tools
 
 repo = Repo(os.getcwd())
 repo.git.remote('add', 'upstream', 'git@github.com:vmware/tern.git')
@@ -25,22 +17,9 @@ repo.git.fetch('upstream')
 hcommit = repo.head.commit
 diff = hcommit.diff('upstream/master')
 
-changes=[]
+if not diff:
+    print('No changes to lint.')
+
 for d in diff:
-    # Get the list of strings for changed files
-    changes.append(d.b_path)
-
-# check that changes has entries
-if not changes:
-    print('No changes to run tests for.')
-    sys.exit(0)
-
-for change in changes:
-    if change[-3:] != '.md':
-        docs_only = False
-        break
-
-if docs_only:
-    sys.exit(1)
-else:
-    sys.exit(0)
+    if os.path.exists(d.b_path) and (d.b_path)[-3:] == '.py':
+        print(d.b_path)

